src/Stratis.Bitcoin.Features.Wallet/MultisigTransactionHandler.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Policy;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Wallet
+{
+    /// <summary>
+    /// A handler that has various functionalities related to transaction operations.
+    /// </summary>
+    public class MultisigTransactionHandler : WalletTransactionHandler, IMultisigTransactionHandler
+    {
+        private readonly ILogger logger;
+
+        private readonly Network network;
+
+        private readonly StandardTransactionPolicy transactionPolicy;
+
+        private readonly IWalletManager walletManager;
+
+        private readonly IWalletFeePolicy walletFeePolicy;
+
+        public MultisigTransactionHandler(
+            ILoggerFactory loggerFactory,
+            IWalletManager walletManager,
+            IWalletFeePolicy walletFeePolicy,
+            Network network,
+            StandardTransactionPolicy transactionPolicy)
+            : base(loggerFactory, walletManager, walletFeePolicy, network, transactionPolicy)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <inheritdoc />
+        public Transaction BuildTransaction(TransactionBuildContext context, SecretModel[] secrets)
+        {
+            if (secrets == null || secrets.Length == 0)
```

Comment:
The private keys should be first looked up in the wallet itself, so we should allow the possibility to pass pubkeys only to the api endpoint and a partially signed trx.